src/eval.c

```diff
@@ -302,11 +328,72 @@ sds luaCreateFunction(client *c, robj *body) {
         return dictGetKey(de);
     }
 
+    /* Handle shebang header in script code */
+    ssize_t shebang_len = 0;
+    if (!strncmp(body->ptr, "#!", 2)) {
```

Comment:
If we're anyway handling this parsing, why not use a Lua comment to keep the entire script pure Lua? I suppose it's not very likely that the same script will be executed as a command line and a Redis script.

Reply:
I think the shebang is a well known concept that we can stick to.
since we don't want these scripts to be able to successfully execute on earlier versions of redis, i think using a Lua command is undesired (if we wanted that would have been a must, and since we don't, then it's not an option).

the other concern is what would happen if bash will find a Lua file with that comment (the non-standard flags we use).
like suggested in this comment: https://github.com/redis/redis/pull/10126#issuecomment-1017692660
i don't think it's a real concern. usually these scripts will contain a `redis.call` calls, that are not compatible with generic Lua anyway.

Reply:
As @zuiderkwast mentioned the `flags=` arguments aren't valid command line arguments to lua anyway so these shebangs can't be used in a command line anyway.
The reason behind the shebang approach and not some comment approach is to make it a standard way to pass metadata about a script independent of the engine. It also makes it possible to select the engine without changing `EVAL`. This is basically the same idea behind why Unix uses shebangs - passing information to the shell independent of the language the script is written in, mainly to select which interpreter to use.
I like this design, but I also see the logic in changing this to a line comment format. But then we'll need another way to specify the engine (if we ever want to support other engines).